alu_u32/src/bitwise/stark.rs

```diff
@@ -0,0 +1,57 @@
+use super::columns::Bitwise32Cols;
+use super::Bitwise32Chip;
+use core::borrow::Borrow;
+use valida_machine::MEMORY_CELL_BYTES;
+
+use p3_air::{Air, AirBuilder};
+use p3_field::AbstractField;
+use p3_matrix::MatrixRows;
+
+impl<F, AB> Air<AB> for Bitwise32Chip
+where
+    F: AbstractField,
+    AB: AirBuilder<F = F>,
+{
+    fn eval(&self, builder: &mut AB) {
+        let main = builder.main();
+        let local: &Bitwise32Cols<AB::Var> = main.row(0).borrow();
+
+        let base_2 = [1, 2, 4, 8, 16, 32, 64, 128].map(AB::Expr::from_canonical_u32);
+
+        for i in 0..MEMORY_CELL_BYTES {
+            let byte_1: AB::Expr = local.bits_1[i]
+                .into_iter()
+                .zip(base_2.iter().cloned())
+                .map(|(bit, base)| bit * base)
+                .sum();
+            let byte_2: AB::Expr = local.bits_2[i]
+                .into_iter()
+                .zip(base_2.iter().cloned())
+                .map(|(bit, base)| bit * base)
+                .sum();
+
+            // Check that input byte decomposition is correct
+            builder.assert_eq(local.input_1[i], byte_1.clone());
+            builder.assert_eq(local.input_2[i], byte_2.clone());
+
+            let bitwise_and: AB::Expr = local.bits_1[i]
+                .into_iter()
+                .zip(local.bits_2[i].into_iter())
+                .map(|(bit_1, bit_2)| bit_1 * bit_2)
+                .sum();
```

Comment:
Seems like we want a weighted sum here, maybe zip with `base_2` to scale the bits?